awx/main/utils/receptor.py

```diff
@@ -7,34 +7,66 @@
 
 logger = logging.getLogger('awx.main.utils.receptor')
 
+__RECEPTOR_CONF = '/etc/receptor/receptor.conf'
+
 
 def get_receptor_sockfile():
-    receptor_conf = '/etc/receptor/receptor.conf'
-    with open(receptor_conf, 'r') as f:
+    with open(__RECEPTOR_CONF, 'r') as f:
         data = yaml.safe_load(f)
     for section in data:
         for entry_name, entry_data in section.items():
             if entry_name == 'control-service':
                 if 'filename' in entry_data:
                     return entry_data['filename']
                 else:
-                    raise RuntimeError(f'Receptor conf {receptor_conf} control-service entry does not have a filename parameter')
+                    raise RuntimeError(f'Receptor conf {__RECEPTOR_CONF} control-service entry does not have a filename parameter')
     else:
-        raise RuntimeError(f'Receptor conf {receptor_conf} does not have control-service entry needed to get sockfile')
+        raise RuntimeError(f'Receptor conf {__RECEPTOR_CONF} does not have control-service entry needed to get sockfile')
+
+
+def get_tls_client(use_stream_tls=None):
+    if not use_stream_tls:
+        return None
+
+    with open(__RECEPTOR_CONF, 'r') as f:
+        data = yaml.safe_load(f)
+    for section in data:
+        for entry_name, entry_data in section.items():
+            if entry_name == 'tls-client':
+                if 'name' in entry_data:
+                    return entry_data['name']
+    return None
 
 
 def get_receptor_ctl():
     receptor_sockfile = get_receptor_sockfile()
-    return ReceptorControl(receptor_sockfile)
+    try:
+        return ReceptorControl(receptor_sockfile, config=__RECEPTOR_CONF, tlsclient=get_tls_client(True))
```

Comment:
it's weird that we put in `tlsclient` to `ReceptorControl` but also gets passed to `submit_work`. I'm a little interested to know more of the details of what the `RuntimeError` here would be.

Reply:
Errors like these shouldn't be hit

https://github.com/ansible/receptor/blob/e8a789ee8c2b684f0368df1f770d338aaf7002a1/receptorctl/receptorctl/socket_interface.py#L21

So I'm skeptical that this try-except is a good idea

Reply:
in what scenarios does a node not have TLS enabled? Is that just so the dev environment works?

Reply:
Then it should fit into the exception where the tlsclient is not specified.  For example, let's consider these 2 `automationcontroller`  instances with and without TLS:

```bash
Node            Service   Type       Last Seen             Tags
192.168.111.121 control   Stream     2021-09-14 20:41:33   {'type': 'Control Service'}
192.168.111.126 control   StreamTLS  2021-09-14 20:42:38   {'type': 'Control Service'}
```
So then that would be:

### awx-manage on `192.168.111.126` box:
```python
>>> ctl = ReceptorControl('/var/run/awx-receptor/receptor.sock', config='/etc/receptor/receptor.conf', tlsclient=get_tls_client(True))
>>> ctl._key
'/etc/receptor/tls/192.168.111.126.key'
```

### awx-manage on `192.168.111.121` box
```python
>>> try:
...     ctl = ReceptorControl('/var/run/awx-receptor/receptor.sock', config='/etc/receptor/receptor.conf', tlsclient=get_tls_client(True))
... except:
...     ctl = ReceptorControl('/var/run/awx-receptor/receptor.sock')
... 
>>> ctl._key
>>> ctl._socketaddress
'/var/run/awx-receptor/receptor.sock'
```

I can spin a lab really quickly if you want to take a look together.

Reply:
Oh, I got it, because it can be turned off :+1:

Reply:
> it's weird that we put in `tlsclient` to `ReceptorControl` but also gets passed to `submit_work`. I'm a little interested to know more of the details of what the `RuntimeError` here would be.

I think we do that because in theory the `submit_work()` can use different `tlsclient` certificate per client (but its not our use case though)